doc/source/cluster/kubernetes/user-guides/resource-isolation-with-writable-cgroups.md

```diff
@@ -0,0 +1,187 @@
+resources for critical system processes. Historically, enabling resource isolation required privileged containers capable of writing to
+the `/sys/fs/cgroup` file system. This approach was not recommended due to the security risks associated with privileged containers. In newer versions of GKE,
+you can enable writable cgroups, granting containers read-write access to the cgroups API without requiring privileged mode.
+
+## Create a GKE Cluster with writable cgroups enabled
+
+To use Ray resource isolation on Kubernetes without privileged containers, you must use a platform that supports cgroupsv2 and writable cgroups.
+
+On GKE, create a cluster with writable cgroups enabled as follows:
+```bash
+$ cat > containerd_config.yaml << EOF
+writableCgroups:
+  enabled: true
+EOF
+
+$ gcloud container clusters create ray-resource-isolation \
+    --cluster-version=1.34 \
+    --machine-type=e2-standard-16 \
+    --num-nodes=3 \
+    --scopes="cloud-platform" \
+    --containerd-config-from-file=containerd_config.yaml
+```
+
+## Create a RayCluster with resource isolation enabled
+
+Create a RayCluster with writable cgroups enabled:
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-cluster-resource-isolation.gke.yaml
+```
+
+The applied manifest enables Ray resource isolation by setting `--enable-resource-isolation` in the `ray start` command. It also includes annotations on the Head and Worker Pods
+to enable writable cgroups, allowing for hierarchical cgroup management within the container:
+```yaml
+metadata:
+  annotations:
+    node.gke.io/enable-writable-cgroups.ray-head: "true"
+```
+
+It also applies a node selector to ensure Ray pods are scheduled only on GKE nodes with this capability enabled:
+```yaml
+nodeSelector:
+  node.gke.io/enable-writable-cgroups: "true"
+```
+
+## System reserved resources
+
+When resource isolation is enabled, Ray manages a cgroup hierarchy within the container:
+
+```text
+       base_cgroup_path (e.g. /sys/fs/cgroup)
+              |
+      ray-node_<node_id>
+     |                 |
+   system             user
+     |               |    |
+    leaf        workers  non-ray
+```
+
+The cgroup hierarchy enables Ray to reserve resources for both system and user-level processes based on the container's total reserved resources.
+
+*   **System-reserved CPU:** By default, Ray reserves between 1 and 3 cores.
+    *   Formula: `min(3.0, max(1.0, 0.05 * num_cores_on_the_system))`
+*   **System-reserved Memory:** By default, Ray reserves between 500MB and 10GB.
+    *   Formula: `min(10GB, max(500MB, 0.10 * memory_available_on_the_system))`
+
+All remaining resources are reserved for user processes (e.g. Ray workers).
+
+## Verify resource isolation for Ray processes
+
+Verify that resource isolation is enabled by inspecting the cgroup filesystem within a Ray container.
+```bash
```

Comment:
This shell output is pretty dense. I'd recommend introducing newlines between subsequent commands for easier reading.